### PR TITLE
Expose a global subroute method.

### PR DESCRIPTION
**Pull request by JorikSchellekens**
#292

### CODE REVIEW

**Comment by glyph** [comment]
Hi @JorikSchellekens ! Thanks for contributing to Klein.  Would you mind updating the docs and/or tests to verify this?

To my mind `subroute` always felt sort of awkward, having to indent huge portions of your code.  Do we really want to expose this in more places or do we want something more like `foo = app.subrouter("/prefix")` ?  I would not block on this concern (we already have the method, the global version should probably be exposed too) but would be interested to hear your thoughts.

**Comment by JorikSchellekens** [comment]
The subroute method is awkward. I had a precursory look at the implementation for it. What's the justification for it's implementation as a context manager?

In the meantime I've added a short bit about it in the docs.

File: docs/examples/subroutes.rst [hunk 1]
```diff
@@ -28,3 +28,17 @@ Here is an example app that has routes for ``/branch/lair``, ``/branch/crypt``,
...
+
+
+The subroute method is also available globally if that is your preferred klein
+pattern.
+
+.. code-block:: python
+
+    from klein import run, subroute
+
+    with subrout("/branch") as app:
```
Comment (wsanchez) [inline docs/examples/subroutes.rst]: subrout ➜ subroute

File: docs/examples/subroutes.rst [hunk 2]
```diff
@@ -28,3 +28,17 @@ Here is an example app that has routes for ``/branch/lair``, ``/branch/crypt``,
...
+      @app.route("/twilightforest")
+      def twighlightforest(request):
+          return b"These stairs lead to the twilight forest."
```
Comment (wsanchez) [inline docs/examples/subroutes.rst]: twighlight or twilight ?

File: docs/examples/subroutes.rst (hunk 1 above)
Comment (JorikSchellekens) [reply docs/examples/subroutes.rst]: done

**Comment by wsanchez** [comment]
Build fail appears to be Travis issues. Kicked.

**Comment by wsanchez** [comment]
OK looks good. @JorikSchellekens sorry about the slow progress, and thanks for hanging in there.

**Pull request merged**